iris-mpc-gpu/src/dot/share_db.rs

```diff
@@ -571,6 +571,87 @@ impl ShareDB {
         }
     }
 
+    pub fn load_host_db_subset_into_chunk_buffers(
+        &self,
+        db: &[Vec<Vec<u16>>],
+        buffers: &DBChunkBuffers,
+        streams: &[CudaStream],
+    ) {
+        // TODO: check if we should do all of them concurrently instead of one by one
```

Comment:
yeah I think moving the synchronize out of the loop should be fine.
We might also want to do the preprocessing of the slices in parallel before, but I think this is fine as a start. Let's add some timings here to see how long preprocessing and copying takes?

Reply:
Atm we cannot do so since the copied data is only allocated inside the loop and will become invalid. Will move them outside and sync after.